client/src/pages/CardInfo-components/CardInfo.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState, useEffect } from "react";
+import {
+  Typography,
+  Container,
+  Box,
+  TextField,
+  Button,
+  CardContent,
+  Card,
+} from "@material-ui/core";
+import { Assignment } from "@material-ui/icons";
+import { useStyles } from "../../themes/cardInfoStyles";
+import CardInfoDescription from "./CardInfoDescription";
+import CardInfoDeadline from "./CardInfoDeadline";
+import CardInfoComment from "./CardInfoComment";
+
+function CardInfo({ task, showCardInfo, closeCardInfo }) {
+  useEffect(() => {
+    console.log(showCardInfo);
+  }, [showCardInfo]);
+
+  const classes = useStyles();
+  const [info, setInfo] = useState({
+    name: "",
+    description: "",
+    deadline: "",
+    comment: "",
+    tags: [],
+    attachments: [],
+  });
+  const [showDescription, setShowDescription] = useState(true);
+  const [showDeadline, setShowDeadline] = useState(true);
+  const [showComment, setShowComment] = useState(true);
+
+  // description section handlers
+  const addDescription = () => {
+    setShowDescription(true);
+  };
+  const handleSaveDescription = (description) => {
+    setInfo({ ...info, description });
+  };
+  const handleDeleteDescription = () => {
+    setShowDescription(false);
+  };
+
+  // deadline section handlers
+  const addDeadline = () => {
+    setShowDeadline(true);
+  };
+  const handleSaveDeadline = (deadline) => {
+    setInfo({ ...info, deadline });
+  };
+  const handleDeleteDeadline = () => {
+    setShowDeadline(false);
+  };
+
+  // comment section handlers
+  const addComment = () => {
+    setShowComment(true);
+  };
+  const handleSaveComment = (comment) => {
+    setInfo({ ...info, comment });
+  };
+  const handleDeleteComment = () => {
+    setShowComment(false);
+  };
+
+  return (
+    <Container
+      maxWidth={false}
+      className={`${classes.bg} ${
+        showCardInfo ? classes.dBlock : classes.dNone
+      }`}
+    >
+      <Card className={classes.cardContainer}>
+        <CardContent className={classes.header}>
+          <Box className={classes.titleContainer}>
```

Comment:
Try using Grid container and item instead of boxes for this

Reply:
My initial implementation used Grid containers and items, but I was unable to properly add an overflow scroll, even with the nowrap prop on the Grid. I managed to get the overflow working with boxes